Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/repair.h

```diff
@@ -186,6 +184,289 @@ degenerate_faces(const TriangleMesh& tm, OutputIterator out)
+{
+  using boost::get_param;
+  using boost::choose_param;
+
+  typedef typename GetVertexPointMap<PolygonMesh, NamedParameters>::const_type VertexPointMap;
+  VertexPointMap vpmap = choose_param(get_param(np, internal_np::vertex_point),
+                                      get_const_property_map(vertex_point, pm));
+  typedef typename GetGeomTraits<PolygonMesh, NamedParameters>::type Traits;
+  Traits traits = choose_param(get_param(np, internal_np::geom_traits), Traits());
+
+  if ( traits.equal_3_object()(get(vpmap, target(e, pm)), get(vpmap, source(e, pm))) )
+    return true;
+}
+
+template <typename PolygonMesh>
+bool is_degenerate_edge(typename boost::graph_traits<PolygonMesh>::edge_descriptor e,
+                        const PolygonMesh& pm)
+{
+  return is_degenerate_edge(e, pm, parameters::all_default());
+}
+
+/// \ingroup PMP_repairing_grp
+/// checks whether a triangle face is degenerate.
+/// A triangle face is considered degenerate if all three points of the face are collinear.
+///
+/// @tparam TriangleMesh a model of `FaceListGraph` and `MutableFaceGraph`
+/// @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
+///
+/// @param f the face to check whether is degenerate
+/// @param tm the triangulated surface mesh upon evaluation
+/// @param np optional \ref pmp_namedparameters "Named Parameters" described below
+///
+/// \cgalNamedParamsBegin
+///    \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `pmesh`. The type of this map is model of `ReadWritePropertyMap`.
+/// If this parameter is omitted, an internal property map for
+/// `CGAL::vertex_point_t` should be available in `PolygonMesh`
+/// \cgalParamEnd
+///    \cgalParamBegin{geom_traits} a geometric traits class instance.
+///       The traits class must provide the nested type `Point_3`,
+///       and the nested functor :
+///         - `Collinear_3` to check whether 3 points are collinear
+///   \cgalParamEnd
+/// \cgalNamedParamsEnd
+///
+/// \return true if the triangle face is degenerate
+template <typename TriangleMesh, typename NamedParameters>
+bool is_degenerate_triangle_face(typename boost::graph_traits<TriangleMesh>::face_descriptor f,
+                                 const TriangleMesh& tm,
+                                 const NamedParameters& np)
+{
+  CGAL_assertion(CGAL::is_triangle_mesh(tm));
+
+  using boost::get_param;
+  using boost::choose_param;
+
+  typedef typename GetVertexPointMap<TriangleMesh, NamedParameters>::const_type VertexPointMap;
+  VertexPointMap vpmap = choose_param(get_param(np, internal_np::vertex_point),
+                                      get_const_property_map(vertex_point, tm));
+  typedef typename GetGeomTraits<TriangleMesh, NamedParameters>::type Traits;
+  Traits traits = choose_param(get_param(np, internal_np::geom_traits), Traits());
+
+  // call from BGL helpers
+  return is_degenerate_triangle_face(f, tm, vpmap, traits);
```

Comment:
remove the implementation from BGL and inline it here. Note you'll have to update a couple of files that were using the old function. General remark, the following `const typename Traits::Point_3& p1 = get(vpm, v)` is not correct, it should be `typename boost::property_traits<Vpm>::reference p1 = get(vpm, v)`.

Reply:
inlining is missing